ecl/eclagent/eclagent.cpp

```diff
@@ -3460,19 +3460,54 @@ extern int HTHOR_API eclagent_main(int argc, const char *argv[], StringBuffer *
             wuXML->kill();  // free up text as soon as possible.
         }
 
+        Owned<IConnectionMonitor> daliDownMonitor;
+
         Owned<IUserDescriptor> standAloneUDesc;
         if (daliServers.length())
         {
+            SocketEndpoint daliEp(daliServers, DALI_SERVER_PORT);
             {
                 MTIME_SECTION(queryActiveTimer(), "SDS_Initialize");
-                Owned<IGroup> serverGroup = createIGroup(daliServers.str(), DALI_SERVER_PORT);
+                Owned<IGroup> serverGroup = createIGroup(1, &daliEp);
                 initClientProcess(serverGroup, DCR_EclAgent, 0, NULL, NULL, MP_WAIT_FOREVER);
             }
 #ifdef MONITOR_ECLAGENT_STATUS  
             serverstatus.reset(new CSDSServerStatus("ECLagent"));
             serverstatus->queryProperties()->setPropInt("Pid", GetCurrentProcessId());
             serverstatus->commitProperties();
 #endif
+            class CDaliDownMonitor : public CSimpleInterfaceOf<IConnectionMonitor>
+            {
+                const SocketEndpoint &daliEp;
+                Semaphore sem;
+            public:
+                CDaliDownMonitor(const SocketEndpoint &_daliEp) : daliEp(_daliEp)
+                {
+                }
+                ~CDaliDownMonitor()
+                {
+                    removeMPConnectionMonitor(this);
+                    sem.signal();
+                }
+                virtual void onClose(SocketEndpoint &ep)
+                {
+                    if (ep.equals(daliEp))
+                    {
+                        WARNLOG("Dali has closed. Waiting 1 minute for process to exit gracefully before killing..");
+
+                        if (sem.wait(60000)) // see if process shutsdown gracefully before killing
+                            return;
+                        WARNLOG("Killing process");
+#ifdef _WIN32
+                        TerminateProcess(GetCurrentProcess(), 1);
+#else
+                        kill(getpid(), SIGKILL);
```

Comment:
just being picky, I prefer raise() if sending to oneself.

Reply:
I don't think we use raise() anywhere at the moment.
Above is copy and paste of code that we use in quite a few place.

I thought about adding a wrapper to remove the platform #ifdef's here, but didn't.
We can do that another time and change to raise()  (so everywhere uses).

Reply:
Seems like this is likely to block the very thread that is trying to close cleanly?

Also, isn't there a guillotine thread running that (a) should have killed eventually anyway and (b) could check periodically that dali connection is still valid.